typescript/infra/src/xerc20/utils.ts

```diff
@@ -0,0 +1,386 @@
+import { BigNumber } from 'bignumber.js';
+import chalk from 'chalk';
+import { PopulatedTransaction } from 'ethers';
+import { join } from 'path';
+
+import { HypXERC20Lockbox__factory } from '@hyperlane-xyz/core';
+import { ChainAddresses } from '@hyperlane-xyz/registry';
+import {
+  ChainMap,
+  EvmXERC20VSAdapter,
+  MultiProtocolProvider,
+  MultiProvider,
+  TokenType,
+  WarpCoreConfig,
+  WarpRouteDeployConfig,
+  canProposeSafeTransactions,
+  isXERC20TokenConfig,
+} from '@hyperlane-xyz/sdk';
+import { Address, CallData, rootLogger } from '@hyperlane-xyz/utils';
+
+import { SafeMultiSend, SignerMultiSend } from '../govern/multisend.js';
+import { getInfraPath } from '../utils/utils.js';
+
+export const XERC20_BRIDGES_CONFIG_PATH = join(
+  getInfraPath(),
+  'scripts/xerc20/config.yaml',
+);
+
+export interface BridgeConfig {
+  xERC20Address: Address;
+  bridgeAddress: Address;
+  decimals: number;
+  owner: Address;
+  bufferCap: number;
+  rateLimitPerSecond: number;
+}
+
+export async function addBridgeToChain({
+  chain,
+  bridgeConfig,
+  multiProtocolProvider,
+  envMultiProvider,
+}: {
+  chain: string;
+  bridgeConfig: BridgeConfig;
+  multiProtocolProvider: MultiProtocolProvider;
+  envMultiProvider: MultiProvider;
+}) {
+  const { xERC20Address, bridgeAddress, bufferCap, rateLimitPerSecond, owner } =
+    bridgeConfig;
+
+  try {
+    const xERC20Adapter = new EvmXERC20VSAdapter(chain, multiProtocolProvider, {
+      token: xERC20Address,
+    });
+
+    const bufferCapBigInt = BigInt(bufferCap);
+    const rateLimitBigInt = BigInt(rateLimitPerSecond);
+
+    const rateLimits = await xERC20Adapter.getRateLimits(bridgeAddress);
+    if (rateLimits.rateLimitPerSecond) {
+      rootLogger.warn(
+        chalk.yellow(
+          `[${chain}] Skipping addBridge as rate limits already set for bridge: ${bridgeAddress}.`,
+        ),
+      );
+      return;
+    }
+
+    const tx = await xERC20Adapter.populateAddBridgeTx({
+      bufferCap: bufferCapBigInt,
+      rateLimitPerSecond: rateLimitBigInt,
+      bridge: bridgeAddress,
+    });
+
+    rootLogger.info(
```

Comment:
Can we have a dry-run flag that just shows the intended transactions without submission?